File: little-bee/controller/router/output.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_OUTPUT = "/config/output"
const URL_ROUTER_OUTPUT_BY_ID = "/config/output/:output-id"
const URL_ROUTER_OUTPUT_QUERY = "/config/output/query"
const URL_ROUTER_OUTPUT_ALL = "/config/output/all"

func OutputRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_OUTPUT, config.CreateOutput)
	r.GET(URL_ROUTER_OUTPUT_BY_ID, config.RetrieveOutput)
	r.PUT(URL_ROUTER_OUTPUT, config.UpdateOutput)
	r.DELETE(URL_ROUTER_OUTPUT_BY_ID, config.DeleteOutput)
	r.PUT(URL_ROUTER_OUTPUT_BY_ID, config.DeleteOutputWithFlag)
	return r
}

File: frontend/console/src/api/output.js
import Axios from "@/http";

export function createOutput(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/output', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveOutput(outputId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/output/${ outputId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateOutput(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/output', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteOutput(outputId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/output/${ outputId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteOutputWithFlag(outputId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/output/${ outputId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllOutputs() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/output/all').then(res => {
            resolve(res)
        })
    })
}

export function queryOutput (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/output/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}